ui/src/components/UploadStory.tsx

```diff
@@ -0,0 +1,855 @@
+      summary: currentStory.summary,
+      author_first_name: currentStory.author_first_name,
+      author_last_name: currentStory.author_last_name,
+      author_country: currentStory.author_country,
+      year: currentStory.year,
+      current_city: currentStory.current_city.toUpperCase(),
+      bio: bio,
+    }
+  );
+
+  const story = React.useMemo(() => {
+    if (isDrawerOpen) {
+      const storyFromData: Story = {
+        image_url:
+          newImage == ""
+            ? URL.createObjectURL(formInput.image)
+            : currentStory.image_url,
+        video_url: formInput.video_url as string,
+        title: formInput.title as string,
+        content: formInput.content as string,
+        summary: formInput.summary as string,
+        author_first_name: formInput.author_first_name as string,
+        author_last_name: formInput.author_last_name as string,
+        author_country: authorCountry as string,
+        year: formInput.year as number,
+        current_city: formInput.current_city as string,
+        author: {
+          ID: 0,
+          CreatedAt: "",
+          UpdatedAt: "",
+          DeletedAt: {
+            Time: "",
+            Valid: true,
+          },
+          firstName: formInput.author_last_name as string,
+          currentCity: formInput.current_city as string,
+          bio: formInput.bio as string,
+        } as Author,
+        tags: [],
+        is_visible: true,
+        latitude: 0,
+        longitude: 0,
+        ID: 0,
+        CreatedAt: "",
+        UpdatedAt: "",
+        DeletedAt: {
+          Time: "",
+          Valid: true,
+        },
+      };
+      return storyFromData;
+    }
+    return undefined;
+  }, [isDrawerOpen, formInput]);
+
+  const hasAllRequiredFields = React.useMemo(() => {
+    return (
+      formInput.image &&
+      formInput.title &&
+      formInput.content &&
+      formInput.summary &&
+      formInput.author_first_name &&
+      formInput.author_last_name &&
+      formInput.year &&
+      formInput.current_city &&
+      formInput.author_country
+    );
+  }, [formInput]);
+
+  const setImage = (files: File[]) => {
+    setFormInput({
+      image: files[0],
+    });
+  };
+
+  const handleKeyDown = (event) => {
+    switch (event.key) {
+      case ",":
+      case " ": {
+        event.preventDefault();
+        event.stopPropagation();
+        if (event.target.value.length > 0) {
+          setTagArrayValues([...tagArray, event.target.value]);
+        }
+        break;
+      }
+      default:
+    }
+  };
+
+  const setErrorUploadingState = (event, reason) => {
+    if (reason === "clickaway") {
+      return;
+    }
+    setErrorOpen(false);
+  };
+
+  const handleChange = (
+    event?: React.ChangeEvent<{ name?: string; value: unknown }>
+  ) => {
+    const name = event.target.name as keyof typeof formInput;
+    setFormInput({
+      [name]: event.target.value,
+    });
+  };
+
+  const handleDialogClose = () => {
```

Comment:
Since we're going to be implementing the new page on successful upload, let's remove all the success dialog code:
`handleDialogClose()`
`handleDialogAgree()`
`hadleDialogDisagree()`

 etc